contentlayer.config.ts

```diff
@@ -1,18 +1,112 @@
-import { defineDocumentType, makeSource, ComputedFields } from "contentlayer2/source-files";
-
-// this is a basic example to show how to structure your content source
-export const _DataName = defineDocumentType(() => ({
-  name: "Post", // name of the post
-  filePathPattern: `**/*.md`, // file path where the mdx file to be process is located
-  contentType: "mdx", // content type mdx
-  // fields: types desc
+import * as fs from "fs";
+import { defineDocumentType, defineNestedType, makeSource } from "contentlayer2/source-files";
+
+const Resources = defineNestedType(() => ({
+  name: "Resources",
   fields: {
-    title: { type: "string", required: true },
-    date: { type: "date", required: true },
-  },
-  computedFields: {
-    url: { type: "string", resolve: (post) => `/posts/${post._raw.flattenedPath}` },
+    title: { type: "string" },
+    url: { type: "string" },
   },
 }));
 
-export default makeSource({ contentDirPath: "posts", documentTypes: [_DataName] });
+export const generateDocuTypes = () => {
+  const getFolders = fs.readdirSync(`${process.cwd()}/public/bitcoin-transcript`, "utf-8");
```

Comment:
we can define a variable for process.cwd since we're using it in several places in this file